src/test/javascript/spec/component/file-upload-assessment/file-upload-assessment-dashboard.component.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { FileUploadSubmissionService } from 'app/exercises/file-upload/participate/file-upload-submission.service';
+import { FileUploadAssessmentsService } from 'app/exercises/file-upload/assess/file-upload-assessment.service';
+
+const course = { id: 1 };
+const testRoute = ({ snapshot: { paramMap: convertToParamMap({ courseId: course.id, exerciseId: 22 }) } } as any) as ActivatedRoute;
+const fileuploadExercise = { id: 22, course, type: ExerciseType.FILE_UPLOAD, studentAssignedTeamIdComputed: true };
+const fileUploadSubmission = { id: 25, submitted: true, result: { id: 10, assessor: { id: 20, guidedTourSettings: [] } } };
+
+describe('FileUploadAssessmentDashboardComponent', () => {
+    let component: FileUploadAssessmentDashboardComponent;
+    let fixture: ComponentFixture<FileUploadAssessmentDashboardComponent>;
+    let fileUploadSubmissionService: FileUploadSubmissionService;
+    let fileUploadAssessmentsService: FileUploadAssessmentsService;
+    let sortService: SortService;
+    let mockAuth: MockAccountService;
+
+    beforeEach(fakeAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [FileUploadAssessmentDashboardComponent],
+            providers: [
+                JhiLanguageHelper,
+                { provide: ActivatedRoute, useValue: testRoute },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .overrideTemplate(FileUploadAssessmentDashboardComponent, '')
+            .compileComponents()
+
+            .then(() => {
+                fixture = TestBed.createComponent(FileUploadAssessmentDashboardComponent);
+                component = fixture.componentInstance;
+                fileUploadAssessmentsService = TestBed.inject(FileUploadAssessmentsService);
+                fileUploadSubmissionService = TestBed.inject(FileUploadSubmissionService);
+                sortService = TestBed.inject(SortService);
+                mockAuth = (fixture.debugElement.injector.get(AccountService) as any) as MockAccountService;
+                mockAuth.hasAnyAuthorityDirect([]);
+                mockAuth.identity();
+                fixture.detectChanges();
+            });
+    }));
+
+    describe('ngOnInit', () => {
+        it('test ngOnInit', fakeAsync(() => {
+            // setup
+            const fileUpServicespy = spyOn(fileUploadSubmissionService, 'getFileUploadSubmissionsForExercise').and.returnValue({ body: { submission: [fileUploadSubmission] } });
+            const getSubSpy = spyOn<any>(component, 'getSubmissions').and.callThrough();
+
+            // call
+            component.ngOnInit();
+            tick();
+
+            // check
+            expect(component.submissions).toEqual([]);
```

Comment:
I do not realy understand what you are testing here? This is the method you want to test:

![image](https://user-images.githubusercontent.com/29718932/100437367-ec29e000-30a0-11eb-80b8-48affc1c34a5.png)

You should test that exercise is set imo.